phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php

```diff
@@ -340,4 +340,142 @@ public function data_install_and_uninstall_fonts() {
 			),
 		);
 	}
+
+	/**
+	 * Tests failure when fonfaces has improper inputs
+	 *
+	 * @covers ::install_fonts
+	 *
+	 * @dataProvider data_install_with_improper_inputs
+	 *
+	 * @param array $font_families     Font families to install in theme.json format.
+	 * @param array $files             Font files to install.
+	 */
+	public function test_install_with_improper_inputs( $font_families, $files = array() ) {
+		wp_set_current_user( self::$admin_id );
+		$install_request    = new WP_REST_Request( 'POST', '/wp/v2/fonts' );
+		$font_families_json = json_encode( $font_families );
+		$install_request->set_param( 'fontFamilies', $font_families_json );
+		$install_request->set_file_params( $files );
+		$response = rest_get_server()->dispatch( $install_request );
+		$this->assertEquals( 400, $response->get_status(), 'Response status is not 400 when font face has both donwload_from_url and uploaded_file properties.' );
```

Comment:
```suggestion
		$this->assertSame( 400, $response->get_status(), 'Response status is not 400 when font face has both donwload_from_url and uploaded_file properties.' );
```

There's an Core initiative to replace all `assertEquals()` with `assertSame()`.